src/main/java/bio/terra/service/common/CreateAzureContainerStep.java

```diff
@@ -49,10 +58,13 @@ public StepResult undoStep(FlightContext context) throws InterruptedException {
     AzureStorageAccountResource datasetAzureStorageAccountResource =
         workingMap.get(getStorageAccountContextKey(), AzureStorageAccountResource.class);
 
-    resourceService.deleteStorageContainer(
-        datasetAzureStorageAccountResource.getResourceId(),
-        profileModel.getId(),
-        context.getFlightId());
+    // If the container was created, delete it
+    if (workingMap.get(SHOULD_PERFORM_CONTAINER_ROLLBACK, Boolean.class)) {
```

Comment:
If the fail occurs before the working map is written this will cause an NPE. If we wanted to prevent that case and not add a `null` check here, it could be fixed by doing the check in a previous step. The same is true of `datasetAzureStorageAccountResource`; if `doStep()` fails before the db is written then `datasetAzureStorageAccountResource.getResourceId()` below will NPE.

It seems like the TDR code has a mix of careful handling for this NPE case and not careful handling. It'd be good to decide what is acceptable for steps and apply the convention consistently to the code.

Reply:
I thought `workingMap.get()` returns null if the value isn't found on the working map?

Reply:
Ah, good call.  I derped that this isn't a primitive and so doesn't default to `false`.  I'll add a null check